### ISSUE TITLE
Refactor API: use interfaces instead of filenames

### ISSUE THREAD

**Issue by slavablind91** [hhrutter/pdfcpu#70]
How about to use this api
```go
func TrimCommand(input io.ReadSeeker, output io.WriteSeeker, pageSelection []string, config *pdf.Configuration) *Command 
```
instead of current
```go
func TrimCommand(pdfFileNameIn, pdfFileNameOut string, pageSelection []string, config *pdf.Configuration) *Command 
```
?

With `reader/writer` possible to use in-memory storage instead of files. Also it's faster than read from disk, and write to disk.

**Comment by hhrutter** [comment]
Hi there,



Right now there are two ways to call into pdfcpu:
* `api.Process(cmd *api.Command)` which is also used by the CLI
* io.ReadSeeker/io.Writer based like in this [example for ReadContext](https://godoc.org/github.com/hhrutter/pdfcpu/pkg/api#pkg-examples)

Also check out these functions in `api.process_test.go` to get started: 
* TestReadSeekerAndWriter (which is really `optimize` using ReadSeeker)
* TestMergeUsingReadSeekerCloser

I am aware this is not clean but it works.
Refactoring the api and providing ReadSeeker across the board makes sense.

**Comment by hhrutter** [comment]
This is fixed with [v0.2](https://github.com/hhrutter/pdfcpu/releases/tag/v0.2)
